### PR TITLE
Feature/fetch class setup

### PR DESCRIPTION
**Pull request by saraho1123**
## Is this a fix or a feature?
feature

## What is the change?
add an APIRequests class to handle GETs, POSTs and DELETEs. 

I attempted chai spies on the POST and DELETE methods, but I cannot get them to work. I am going to need to reach out for help. I think I do have them set up in the class correctly though.

## What does it fix?
n/a

## Where should the reviewer start?
line 1 of Fetch.js
line 1 of fetch-test.js

## How should this be tested?
run npm test in terminal (2nd and 3rd tests do not yet pass)

## Relevant Tickets:
#13

### CODE REVIEW

File: src/Fetch.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+
+class APIRequests {
+  constructor() {
+    this.urlRoot = 'https://fe-apps.herokuapp.com/api/v1/overlook/1904/'
```
Comment (atownse) [inline src/Fetch.js:4]: Interesting approach with a class

File: src/Fetch.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        'data': inputDate,
+        'roomNumber': inputRoomNumber,
+      })
+    })
+    .then(response => response.json())
+    .then(data => console.log(data))
+    .catch(error => console.log(error))
+  }
+
+  deleteData(urlPath, inputID) {
```
Comment (atownse) [inline src/Fetch.js:31]: Will this be extended to manager?

File: test/fetch-test.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+const expect = chai.expect;
+const APIRequests = require('../src/Fetch');
+
+describe('APIRequest', function() {
+
+  before(function() {
+    global.APIRequests = {};
+    chai.spy.on(APIRequests, ['fetchData'], () => {});
+  });
+
```
Comment (atownse) [inline test/fetch-test.js:15]: May want an afterEach

**Review by atownse — APPROVED** [review]

File: src/Fetch.js (hunk 2 above)
Comment (saraho1123) [reply src/Fetch.js:31]: Oh! I hadn't thought of that. I am not sure.

File: test/fetch-test.js (hunk 3 above)
Comment (saraho1123) [reply test/fetch-test.js:15]: I am not sure what an afterEach does. I will have to look it up. Thanks for the tip!

**Pull request merged**